proto/umee/incentive/v1/query.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+package umeenetwork.umee.incentive.v1;
+
+import "cosmos/base/query/v1beta1/pagination.proto";
+import "google/api/annotations.proto";
+import "gogoproto/gogo.proto";
+import "umee/incentive/v1/incentive.proto";
+
+option go_package = "github.com/umee-network/umee/v3/x/incentive";
+
+// Query defines the gRPC querier service.
+service Query {
+  // Params queries the parameters of the x/incentive module.
+  rpc Params(QueryParams) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/umee/incentive/v1/params";
+  }
+
+  // CompletedIncentivePrograms queries for all incentives programs that have been passed
+  // by governance, and either run to completion or expired immediately due to not being funded.
+  rpc CompletedIncentivePrograms(QueryCompletedIncentivePrograms)
+      returns (QueryCompletedIncentiveProgramsResponse) {
+    option (google.api.http).get = "/umee/incentive/v1/incentive_programs/completed";
+  }
+
+  // OngoingIncentivePrograms queries for all incentives programs that have been passed
+  // by governance, funded, and started but not yet completed.
+  rpc OngoingIncentivePrograms(QueryOngoingIncentivePrograms)
+      returns (QueryOngoingIncentiveProgramsResponse) {
+    option (google.api.http).get = "/umee/incentive/v1/incentive_programs/ongoing";
+  }
+
+  // UpcomingIncentivePrograms queries for all incentives programs that have been passed
+  // by governance, but not yet started. They may or may not have been funded.
+  rpc UpcomingIncentivePrograms(QueryUpcomingIncentivePrograms)
+      returns (QueryUpcomingIncentiveProgramsResponse) {
+    option (google.api.http).get = "/umee/incentive/v1/incentive_programs/upcoming";
+  }
```

Comment:
Frontend will probably just need current, and it's easier to have them stored separately in state since it reduces iterations when looking for just one type.

Reply:
1. So you are moving the program between storage prefixes?
2. My question was more related to the query, why having 3 queries if we can only have 1 or 2? That would reduce and simplify the code. 
3. Let's verify with the frontend. My gut feeling is that they will like to show both present and future.